pay-queue/src/pay_queue/external/gcp_auth.py

```diff
@@ -0,0 +1,46 @@
+# pylint: skip-file
+# flake8: noqa
+# This will get moved to an external library, which is linted by black (different than our rules)
+"""Move this to external library."""
+import functools
+from http import HTTPStatus
+
+import google.oauth2.id_token as id_token
+from cachecontrol import CacheControl
+from flask import abort, current_app, request
+from google.auth.transport.requests import Request
+from requests.sessions import Session
+
+
+def verify_jwt(session):
+    """Check token is valid with the correct audience and email claims for configured email address."""
+    try:
+        jwt_token = request.headers.get('Authorization', '').split()[1]
+        claims = id_token.verify_oauth2_token(
+            jwt_token,
+            Request(session=session),
+            audience=current_app.config.get('PAY_SUB_AUDIENCE')
+        )
+        # Check if the email is verified and matches the configured email
+        required_email = current_app.config.get('VERIFY_PUBSUB_EMAIL')
+        if not claims.get('email_verified') or claims.get('email') != required_email:
+            return 'Email not verified or does not match', 401
+    except Exception as e:
+        return f'Invalid token: {e}', 400
+    return None
+
+
+def ensure_authorized_queue_user(f):
+    """Ensures the user is authorized to use the queue."""
+    @functools.wraps(f)
+    def decorated_function(*args, **kwargs):
+        # Use CacheControl to avoid re-fetching certificates for every request.
```

Comment:
Comment should go down to line 43. Might be worth taking some of this stuff out if it already works..

Reply:
fixed